### ISSUE TITLE
JavaScript package template fails to build on Linux

### ISSUE THREAD

**Issue by kj** [DeterminateSystems/zero-to-nix#184]
Unfortunately as a result of [this fix](https://github.com/DeterminateSystems/zero-to-nix/pull/174) I’m now seeing this error from `nix build`:

```
npm ERR! [esbuild] Failed to find package "esbuild-linux-64" on the file system
```

I think that this is because package-lock.json was generated on a macOS system so it only includes `esbuild-darwin-arm64` and not `esbuild-linux-64` (or other architectures).

**Comment by lucperkins** [comment]
@kj Ah, great catch! I really dislike that tools like npm and Rubygems do this. I'll see if there's a way around it.

**Comment by lucperkins** [comment]
I _think_ that I addressed this in #208. If you could try again and provide feedback, that'd be awesome. I'll reopen this issue if it still doesn't work for you on Linux.

**Comment by kj** [comment]
Thanks @lucperkins, that's working perfectly for me now!
